src/utils.js

```diff
@@ -55,33 +55,77 @@ const OPTION_GETTERS = {
     ruleValueToPrettierOption: getArrowParens
   }
 };
-
+const PARSER_EXTENSION_MAPPING = {
+  css: 'css',
+  gql: 'graphql',
+  html: 'html',
+  js: 'babel',
+  json: 'json',
+  jsx: 'babel',
+  less: 'less',
+  md: 'markdown',
+  mjs: 'babel',
+  scss: 'scss',
+  tsx: 'typescript',
+  ts: 'typescript'
+};
 /* eslint import/prefer-default-export:0 */
 export { getESLintCLIEngine, getOptionsForFormatting, requireModule };
 
 function getOptionsForFormatting(
   eslintConfig,
   prettierOptions = {},
   fallbackPrettierOptions = {},
-  eslintPath
+  eslintPath,
+  fileExtension
 ) {
-  const eslint = getRelevantESLintConfig(eslintConfig, eslintPath);
+  const eslint = getRelevantESLintConfig({
+    eslintConfig,
+    eslintPath,
+    fileExtension
+  });
+
   const prettier = getPrettierOptionsFromESLintRules(
     eslintConfig,
     prettierOptions,
     fallbackPrettierOptions
   );
+  prettier.parser = getPrettierOptionsParser(
+    prettierOptions.parser,
+    fallbackPrettierOptions.parser,
+    eslint.parser,
+    fileExtension
+  );
+
   return { eslint, prettier };
 }
 
-function getRelevantESLintConfig(eslintConfig, eslintPath) {
+function getESLintConfigParser(parser, fileExtension) {
+  if (parser) {
+    return parser;
+  }
+
+  if (['.ts', '.tsx'].includes(fileExtension)) {
+    return require.resolve('@typescript-eslint/parser');
+  }
+
+  if (['.vue'].includes(fileExtension)) {
+    return require.resolve('vue-eslint-parser');
+  }
+  return undefined;
+}
+
+function getRelevantESLintConfig({ eslintConfig, eslintPath, fileExtension }) {
   const cliEngine = getESLintCLIEngine(eslintPath);
   // TODO: Actually test this branch
   // istanbul ignore next
   const loadedRules =
     (cliEngine.getRules && cliEngine.getRules()) ||
     // XXX: Fallback list of unfixable rules, when using and old version of eslint
-    new Map([['global-require', { meta: {} }], ['no-with', { meta: {} }]]);
+    new Map([
```

Comment:
Prettier 2 did this. I attempted to fix with ESLint rules but couldn't find the right settings. This could be a breaking change. I'll have to look at the Prettier 2 code to see what it defaults to. Looking at the `[changelog](https://prettier.io/blog/2020/03/21/2.0.0.html)` for Prettier 2, they didn't list this as a change.